src/App.tsx

```diff
@@ -1,29 +1,35 @@
-import { Component } from 'react';
+import React, { useState, useMemo } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 
-interface State {
-  movies: Movie[];
-}
+export const App: React.FC = () => {
+  const [movies, setMovies] = useState<Movie[]>([]);
 
-export class App extends Component<{}, State> {
-  state: State = {
-    movies: [],
-  };
+  const addMovie = useMemo(() => {
+    return (Movie: Movie) => {
+      setMovies(currentMovies => {
+        if (currentMovies.some((movie: Movie) => movie.imdbID === Movie.imdbID)) {
+          return currentMovies;
+        }
 
-  render() {
-    const { movies } = this.state;
+        currentMovies.push(Movie);
```

Comment:
```suggestion
        currentMovies = [ ...currentMovies, Movie ];
```
Стейт мутувати не можна